packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -151,6 +157,28 @@ class DataStore implements IDataStore {
         return this.fluidDataStoreChannel.request(request);
     }
 
+    /**
+     * Exposes a handle to the data store's root object / entrypoint. Use this as the primary way of interacting with
+     * the data store, and only fall back to requesting the root object through the request pattern if this property
+     * or the handle within it are not defined.
+     */
+    public get IFluidLoadable(): IFluidLoadable | undefined {
+        // While we plumb entrypoints everywhere and this way of getting to the data store's entrypoint could still be
+        // undefined, we have to do some sleight-of-hand and return an object whose 'handle' property is not undefined
+        // so it matches the definition of the IFluidLoadable interface.
+        const maybeIFluidLoadable: IFluidDataStoreChannel & FluidObject<IProvideFluidLoadable>
+            = this.fluidDataStoreChannel;
+        const handle = maybeIFluidLoadable.IFluidLoadable?.handle;
+        if (handle !== undefined) {
+            return {
+                handle,
+                get IFluidLoadable() {
+                    return this;
+                },
```

Comment:
Why do we need this dance with wrapping handle? Can we return handle everywhere? If IFluidLoadable is required, why not return original one? I do not think we provide same promises as exists in COM/native side (RE having guarantees that one gets same pointer when asking for same interface).

I think more important note - DataStore is not "loadable" object. It acts like proxy for something else. So either it needs to return a handle (with a clear name on what this handle represents), or it needs to have an explicit method that returns IFluidLoadable, and the name of that method should clearly indicate what this IFluidLoadable represents.
While latter is more correct, it will make code usage even uglier, so please shift toward using handles :)